addons/pos_restaurant/static/src/app/floor_screen/floor_screen.js

```diff
@@ -28,19 +31,85 @@ export class FloorScreen extends Component {
         const floor = this.pos.currentFloor;
         this.state = useState({
             selectedFloorId: floor ? floor.id : null,
+            floorHeight: "100%",
+            floorWidth: "100%",
             selectedTableIds: this.getTablesSelectedByDefault(),
             isColorPicker: false,
         });
         this.floorMapRef = useRef("floor-map-ref");
+        this.floorScrollBox = useRef("floor-map-scroll");
         const ui = useState(useService("ui"));
         const mode = localStorage.getItem("floorPlanStyle");
         this.pos.floorPlanStyle = ui.isSmall || mode == "kanban" ? "kanban" : "default";
         this.map = useRef("map");
-        onMounted(() => this.pos.openCashControl());
+        onMounted(() => {
+            this.pos.openCashControl();
+        });
         onWillStart(this.onWillStart);
+        useEffect(
+            () => {
+                this.computeFloorSize();
+            },
+            () => [this.activeFloor, this.pos.floorPlanStyle]
+        );
+        useEffect(
+            () => {
+                document.addEventListener("resize", this.computeFloorSize);
+                return () => {
+                    document.removeEventListener("resize", this.computeFloorSize);
+                };
+            },
+            () => []
+        );
+        useEffect(
+            () => {
+                if (hasTouch()) {
+                    if (this.state.selectedTableIds.length) {
+                        this.floorScrollBox.el.classList.remove("overflow-scroll");
+                        this.floorScrollBox.el.classList.add("overflow-hidden");
+                    } else {
+                        this.floorScrollBox.el.classList.remove("overflow-hidden");
+                        this.floorScrollBox.el.classList.add("overflow-scroll");
+                    }
+                }
+            },
+            () => [this.state.selectedTableIds]
+        );
+    }
+    computeFloorSize() {
+        if (this.pos.floorPlanStyle === "kanban") {
+            this.state.floorHeight = "100%";
+            this.state.floorWidth = window.innerWidth + "px";
+            return;
+        }
+        const tables = this.activeFloor.table_ids;
+        const floorV = this.floorMapRef.el.clientHeight;
+        const floorH = this.floorMapRef.el.clientWidth;
+        const positionH = Math.max(
+            ...tables.map((table) => table.position_h + table.width),
+            floorH
+        );
+        const positionV = Math.max(
+            ...tables.map((table) => table.position_v + table.height),
+            floorV
+        );
+
+        if (this.activeFloor.floor_background_image) {
+            const i = new Image();
```

Comment:
```suggestion
            const img = new Image();
```
use a more sensible variable name